src/go/rpk/pkg/cli/transform/init.go

```diff
@@ -0,0 +1,257 @@
+/*
+* Copyright 2023 Redpanda Data, Inc.
+*
+* Use of this software is governed by the Business Source License
+* included in the file licenses/BSL.md
+*
+* As of the Change Date specified in that file, in accordance with
+* the Business Source License, use of this software will be governed
+* by the Apache License, Version 2.0
+ */
+
+package transform
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/transform/buildpack"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/transform/project"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/transform/template"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type optionalBool struct {
+	ok *bool
+}
+
+func (b *optionalBool) Set(s string) error {
+	if strings.ToLower(s) == "prompt" {
+		b.ok = nil
+		return nil
+	}
+	v, err := strconv.ParseBool(s)
+	b.ok = &v
+	return err
+}
+
+func (*optionalBool) Type() string {
+	return "bool"
+}
+
+func (b *optionalBool) String() string {
+	if b.ok == nil {
+		return "prompt"
+	}
+	return strconv.FormatBool(*b.ok)
+}
+
+func (*optionalBool) IsBoolFlag() bool { return true }
+
+func newInitializeCommand(fs afero.Fs) *cobra.Command {
+	var (
+		name    string
+		lang    project.WasmLang
+		install optionalBool
+	)
+	cmd := &cobra.Command{
+		Use:   "init [DIRECTORY]",
+		Short: "Initialize a transform",
+		Long: `Initialize a transform.
+
+Creates a new transform using a template in the current directory.
+
+A new directory can be created by specifying it in the command, like:
+
+rpk transform init foobar
+
+Will initialize a transform project in the foobar directory.
+		`,
+		Args: cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			path, err := computeProjectDirectory(fs, args)
+			out.MaybeDie(err, "unable to determine project directory: %v", err)
+			c := filepath.Join(path, project.ConfigFileName)
+			ok, err := afero.Exists(fs, c)
+			out.MaybeDie(err, "unable to determine if %q exists: %v", c, err)
+			if ok {
+				out.Die("there is already a transform at %q, please delete it before retrying", c)
+			}
+			for name == "" {
```

Comment:
[question] Why a while loop instead of an if here?

Reply:
If the user just presses enter without typing (and there is no suggestion) we'll get an empty string back. Should I:
- Add a comment
- Just bail if the user input is the empty string

Reply:
Oh I see, so it will keep asking the user to enter a name. :thinking: 

I'm fine with that, but also fine with `Just bail if the user input is the empty string`. So, I'll leave it up to you. Thanks!

Reply:
Most cases there is a suggestion so I think I'll leave it.

Reply:
We usually bail in our CLI today if the user enters invalid input, fwiw.

Reply:
Cool - I will open a PR tomorrow to switch it to bail then 👍

Reply:
https://github.com/redpanda-data/redpanda/pull/13965